### DISCUSSION TITLE
Allow Admins to Siege

### DISCUSSION THREAD

**Discussion by laceyj2** [GriefPrevention/GriefPrevention#1182] [General]
Admins currently are unable to siege at all. Admins on my server are a tight nit group and trusted not to abuse powers.

**Comment by RoboMWM** [comment]
I'm not sure how people are creating issues without a template.

**Reply by laceyj2** [reply]
Clicked issue button ->prompted to make account->made account->backpage to issue button->create post

**Reply by RoboMWM** [reply]
Interesting, thanks.

**Comment by RoboMWM** [comment]
Anyways, check permissions in wiki?

**Reply by laceyj2** [reply]
I did,  unfortunately because admin is OP'ed they have all perms including "griefprevention.siegeimmune" and i cant figure out a way to stop them from having that immunity without using /deop and removing "griefprevention.admin.*" from LP. ive searched for some time on forums and google and havent found anything regarding my issue specifically.

I guess i could just add every single plug-in permission one by one to admin roles minus the siegeimmune one but that is beyond tedious.

**Reply by Jikoo** [reply]
There are a bunch of ways around that. You can negate the node using your permissions plugin for your admin group, you can create a default node using Bukkit's permissions.yml feature that will cause the node to be inherited as false, etc.

It depends on your use case, but in my prior management of a small server I can say without a doubt deopping all admins and manually setting up permissions was one of the best moves I made. It gave me a lot of peace of mind to know for sure that no one would accidentally `/stop` the server (again) while I was off at work. Nothing says "fun afternoon" like spending your break downloading a ssh client to your phone to stop people spamming your DMs about the server being broken. Using op/wildcard nodes as a substitute for a real permissions setup is lazy and makes your life harder longterm with issues like this. It also makes it way harder when developers are trying to track down permissions problems that are almost always user error. Manually adding plugin permissions does suck, but the fine grain control you get is (in my opinion) worth the time investment.

**Reply by laceyj2** [reply]
Ill look up some guides on getting  the node negated in LP. Im really new to this so i appreciate your reply.

**Reply by Jikoo** [reply]
For LP, I believe it's as simple as `/perm group <group> permission set griefprevention.siegeimmune false`, but if that's wrong LP's tab completion is phenomenal and should set you right.

If you're looking to use [permissions.yml](https://bukkit.gamepedia.com/Permissions.yml), you can include a section like this:
```yaml
my.custom.node:
  default: op
  children:
    griefprevention.siegeimmune: false
```
Since ops will be granted `my.custom.node` it will then remove the siege immunity node.

**Reply by laceyj2** [reply]
This is amazing thank you so much. I already have gone into the process of deoping admins and setting up permissions properly but this knowledge will definitely help out in the future.

**Comment by Lollipop56-UniverseMC** [comment]
Just do /lp group admin permission set griefprevention.siegeimmune false

Then you're all set.
